holmes/utils/holmes_sync_toolsets.py

```diff
@@ -0,0 +1,155 @@
+import yaml
+from holmes.core.supabase_dal import SupabaseDal
+from holmes.plugins.toolsets import load_builtin_toolsets
+from holmes.core.tools import get_matching_toolsets
+from holmes.common.env_vars import ENABLED_BY_DEFAULT_TOOLSETS, CLUSTER_NAME
+import os
+from pydantic import ValidationError
+from holmes.core.tools import ToolsetYamlFromConfig, ToolsetDBModel, YAMLToolset, ToolsetTag
+from holmes.plugins.prompts import load_and_render_prompt
+from holmes.utils.definitions import CUSTOM_TOOLSET_LOCATION
+import logging
+from datetime import datetime
+
+
+def load_custom_toolsets_config() -> list[ToolsetYamlFromConfig]:
+    """
+    Loads toolsets config from /etc/holmes/config/custom_toolset.yaml with ToolsetYamlFromConfig class
+    that doesn't have strict validations. 
+    Example configuration:
+
+    kubernetes/logs:
+        enabled: false
+  
+    test/configurations:
+        enabled: true
+        icon_url: "example.com"
+        description: "test_description"
+        docs_url: "https://docs.docker.com/"
+        variables:
+            api_endpoint: "$API_ENDPOINT"
+        prerequisites:
+            - env:
+                - API_ENDPOINT
+            - command: "curl ${API_ENDPOINT}"
+        additional_instructions: "jq -r '.result.results[].userData | fromjson | .text | fromjson | .log'"
+        tools:
+            - name: "curl_example"
+            description: "Perform a curl request to example.com using variables"
+            command: "curl -X GET '{{api_endpoint}}?query={{ query_param }}' "
+    """
+    loaded_toolsets = []
+    if os.path.isfile(CUSTOM_TOOLSET_LOCATION):
+        with open(CUSTOM_TOOLSET_LOCATION) as file:
+            parsed_yaml = yaml.safe_load(file)
+            toolsets = parsed_yaml.get("toolsets", {})
+            for name, config in toolsets.items():
+                try:
+                    validated_config = ToolsetYamlFromConfig(**config, name=name)
+                    validated_config.set_path(CUSTOM_TOOLSET_LOCATION)
+                    loaded_toolsets.append(validated_config)
+                except ValidationError as e:
+                    logging.error(f"Toolset '{name}' is invalid: {e}")
+    return loaded_toolsets
+
+
+def merge_and_override_bultin_toolsets_with_toolsets_config(
+    toolsets_loaded_from_config: list[ToolsetYamlFromConfig],
+    default_toolsets_by_name: dict[str, YAMLToolset],
+    enabled_by_default_toolsets: list[YAMLToolset],
+) -> dict[str, YAMLToolset]:
+    """
+    Merges and overrides default_toolsets_by_name with custom 
+    config from /etc/holmes/config/custom_toolset.yaml
+    """
+    toolsets_with_updated_statuses = {}
```

Comment:
why do we need this code?
Can't we get that logic free from helm?
Since the toolsets are dictionaries, wouldn't helm solve the merge for us?

Reply:
@arikalon1 This particular code is not related to helm logic. It loads both python and yaml toolsets from `holmes/plugins/toolsets` and updates them with helm config